code/game/objects/structures/crates_lockers/closets/secure/personal.dm

```diff
@@ -43,13 +43,26 @@
 		to_chat(user, "<span class='warning'>Invalid identification card.</span>")
 		return
 
+	if(opened)
+		to_chat(user, "<span class='notice'>Close the locker first.</span>")
+		return
+
+	if(broken)
+		to_chat(user, "<span class='warning'>The locker appears to be broken.</span>")
+		return
+
+	if(user.loc == src)
+		to_chat(user, "<span class='notice'>You can't reach the lock from inside.</span>")
+		return
```

Comment:
and this, do you think it should be before or after the invalid id check? The closet would tell you if the id is correct if you are inside of it and therefore cant reach it?

Reply:
Moved the invalid ID check all the way up, definitely makes more sense that it tells you the lock is broken/unreachable before it tells you that you're using an invalid ID type.